device/sycl/src/seeding/seed_finding.sycl

```diff
@@ -294,31 +281,98 @@ vecmem::data::vector_buffer<seed> seed_finding::operator()(
     auto weightUpdatingRange = traccc::sycl::calculate1DimNdRange(
         triplet_prefix_sum.size(), weightUpdatingLocalSize);
 
+    // Check if device is capable of allocating sufficient local memory
+    assert(sizeof(scalar) * m_seedfilter_config.compatSeedLimit *
+               weightUpdatingLocalSize <
+           details::get_queue(m_queue)
+               .get_device()
+               .get_info<::sycl::info::device::local_mem_size>());
+
     // Update the weight of all of the spacepoint triplets.
     details::get_queue(m_queue)
         .submit([&](::sycl::handler& h) {
+            // Array for temporary storage of triplets for comparing within seed
+            // selecting kernel
+            ::sycl::accessor<scalar, 1, ::sycl::access::mode::read_write,
+                             ::sycl::access::target::local>
+                local_mem(m_seedfilter_config.compatSeedLimit *
+                              weightUpdatingLocalSize,
+                          h);
+
             h.parallel_for<kernels::update_triplet_weights>(
-                weightUpdatingRange, [filter_config = m_seedfilter_config,
-                                      g2_view, triplet_prefix_sum_view,
-                                      triplet_view](::sycl::nd_item<1> item) {
+                weightUpdatingRange,
+                [filter_config = m_seedfilter_config, g2_view,
+                 triplet_prefix_sum_view, local_mem,
+                 triplet_view](::sycl::nd_item<1> item) {
+                    // Each thread uses compatSeedLimit elements of the array
+                    scalar* dataPos =
+                        &local_mem.get_pointer()[item.get_local_id() *
```

Comment:
I wonder if we should rather skip on using `get_pointer()` here. :thinking: Wouldn't something like this work?

```c++
scalar* data = &(local_mem[item.get_local_id() * filter_config.compatSeedLimit]);
```

Maybe it doesn't work exactly like that, but SYCL accessors should have a way to get the element that we're after without using `get_pointer()`... :thinking: